hack/chains/chains.yaml

```diff
@@ -0,0 +1,24 @@
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: v1
+kind: TektonResourcePlugin
```

Comment:
What's this?

Reply:
In kustomize one can have plugins, this one is called [exec KRM function](https://kubectl.docs.kubernetes.io/guides/extending_kustomize/exec_krm_functions/), the `kind` and most other fields are ceremonial here, what's important is that we can now invoke a shell script and dynamically contribute objects (YAML files) to the set of objects Kustomize has defined statically. Kustomize is very intentionally anti-templating, so we can't point to a resource with `${version}` as a variable within its URL. Note that this won't work with `kubectl kustomize` as it explicitly doesn't include the exec plugins feature. What we arrive at with this is that we have a Kustomize-pure solution, making it easy to be incorporated in acceptance tests, i.e. we can use Kustomize's go API and point to a directory here and have the same environment provisioned from go in addition to shell script. Rather neat I would say.